crates/ironrdp-rdpdr/src/pdu/esc.rs

```diff
@@ -497,12 +699,34 @@ pub mod rpce {
             Ok(())
         }
 
+        fn decode(src: &mut ReadCursor<'_>) -> PduResult<Self> {
+            ensure_size!(in: src, size: Self::size());
+            let version = src.read_u8();
+            let endianness = Endianness::try_from(src.read_u8())?;
+            let common_header_length = src.read_u16();
+            let filler = src.read_u32();
+            if endianness == Endianness::LittleEndian {
+                Ok(Self {
+                    version,
+                    endianness,
+                    common_header_length,
+                    filler,
+                })
+            } else {
+                Err(invalid_message_err!(
+                    "decode",
+                    "StreamHeader",
+                    "server returned big-endian data, parsing not implemented"
+                ))
+            }
+        }
+
         fn size() -> usize {
-            8
+            size_of::<u8>() + size_of::<u8>() + size_of::<u16>() + size_of::<u32>()
```

Comment:
The more I see this kind of code, the more I feel there is something better to do:
```suggestion
            1 /* Version */ + 1 /* Endianness */ + 2 /* CommonHeaderLength */ + 4 /* Filler */
```
It’s easier to double check with the Microsoft documentation:
![image](https://github.com/Devolutions/IronRDP/assets/3809077/681a83c9-aa83-4f48-85f4-0721c3148508)

And I don’t feel like `size_of::<u8>()` is really useful, it’s not like the size of `u8` was going to change and was hard to predict (that would be a problem for a network protocol in the first place). The name of the field is much more useful in my opinion.

I’m not asking that you change this everywhere immediately now, but I wanted to point that out and maybe change our approach for new code in the future.

cc @pacmancoder